apps/ios/GuideDogs/Code/Data/Authored Activities/AuthoredActivityContent.swift

```diff
@@ -225,24 +233,25 @@ extension AuthoredActivityContent {
                                            pois: [])
             
         case "2":
-            guard let route = gpx.routes.first, route.routepoints.count > 0 else {
+            guard let route = gpx.routes.first, route.points.count > 0 else {
                 return nil
             }
             
-            let wpts: [ActivityWaypoint] = waypoints(from: route.routepoints)
+            let wpts: [ActivityWaypoint] = waypoints(from: route.points)
             
             // For waypoints in this experience, require names, descriptions, and street addresses
             guard !wpts.isEmpty, !wpts.contains(where: { $0.name == nil }) else {
                 return nil
             }
             
-            let pois = gpx.waypoints.map { ActivityPOI(coordinate: $0.coordinate, name: $0.name, description: $0.desc) }
+            // TODO: maybe don't use !
```

Comment:
What's this comment about? `pois` is used in the returned value below.

Reply:
Basically, since many of the waypoint properties are now optional, I replaced
```swift
let pois = gpx.waypoints.map { ActivityPOI(coordinate: $0.coordinate, name: $0.name, description: $0.desc) }
```
with
```swift
let pois = gpx.waypoints.map { ActivityPOI(coordinate: $0.coordinate!, name: $0.name!, description: $0.desc) }
```
(which gets cut off immediately after the comment).

The comment is basically saying that I'm not sure what we should do if `$0.name` is nil so I just used `$0.name!` to assert that it's not
(which means that loaded GPX files may break it if their waypoints don't include name and coordinate).

#### Possible Replacements
1. Reject the entire `AuthoredActivityContent` if any of the waypoints are missing these properties
2. Skip the waypoint if it is missing these properties (this may have strange results as waypoints are skipped)
3. Pick some kind of defaults (probably less useful than option 2)

Reply:
Ah, you meant don't use a literal exclamation point. I was confused why you appeared to be scolding yourself, vaguely. :)

Yes, I agree we should pick an option that doesn't induce a crash when the data format doesn't meet our expectations. For `coordinate`, I like option \#2, but for `name`, option \#3 would be better, since it avoids (literally) silently dropping data. We expect this shouldn't really happen in practice, so using something clunky like "Unlabeled point" as a default value might be fine.

Reply:
It looks like the original developers preferred option 1, since there are guards right above this that do the following:
```swift
// For waypoints in this experience, require names, descriptions, and street addresses
guard !wpts.isEmpty, !wpts.contains(where: { $0.name == nil }) else {
    return nil
}
```
so I'm just going to expand these to also check coordinate as well

Reply:
Alright, sounds good!